File: src/main/java/com/ircserv/inter/ServerInterface.java
package com.ircserv.inter;

import com.ircserv.metier.Message;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;

public interface ServerInterface extends Remote{

    ArrayList<Message> getMessages() throws RemoteException;
    ArrayList<Message> getMessages(int nbLastMessage) throws RemoteException;
    void send(String pseudo, String message) throws RemoteException;
    void uploadFile(String pseudo, byte[] data, String filename) throws RemoteException;
}

File: src/inter/ServerInterface.kt
package inter

import metier.Message

import java.rmi.Remote
import java.rmi.RemoteException
import java.util.ArrayList

interface ServerInterface : Remote {


    val messages: ArrayList<Message>
        @Throws(RemoteException::class) get(){
            return this.messages
        }


    @Throws(RemoteException::class)
    fun send(pseudo: String, message: String)
}
